### PR TITLE
No longer do tracing within searcher

### PR DESCRIPTION
**Pull request by ijt**
This is meant to help with #5215.

These traces apparently happen for every repo being searched, but there can be thousands of repos and that would overwhelm lightstep.

Test plan: run this within an instance with say 10k repositories and see if the lightstep traces are usable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ All notable changes to Sourcegraph are documented in this file.
 ### Changed
 
 - A `hardTTL` setting was added to the [Bitbucket Server `authorization` config](https://docs.sourcegraph.com/admin/external_service/bitbucketserver#configuration). This setting specifies a duration after which a user's cached permissions must be updated before any user action is authorized. This contrasts with the already existing `ttl` setting which defines a duration after which a user's cached permissions will get updated in the background, but the previously cached (and now stale) permissions are used to authorize any user action occuring before the update concludes. If your previous `ttl` value is larger than the default of the new `hardTTL` setting (i.e. **3 days**), you must change the `ttl` to be smaller or, `hardTTL` to be larger.
+- Searcher no longer makes spans, to lighten the load on Lightstep so its traces can be viewed. [#5303](https://github.com/sourcegraph/sourcegraph/pull/5303)
```
Comment (keegancsmith) [inline CHANGELOG.md]: Is this worth communicating to our users?

File: cmd/searcher/search/eval.go [hunk 2]
```diff
@@ -32,7 +32,6 @@ import (
 	"github.com/sourcegraph/sourcegraph/pkg/search/matchtree"
 	"github.com/sourcegraph/sourcegraph/pkg/search/query"
 	"github.com/sourcegraph/sourcegraph/pkg/store"
-	"golang.org/x/net/trace"
```
Comment (keegancsmith) [inline cmd/searcher/search/eval.go]: this is unrelated to lightstep. You probably wanna keep this.

Edit: This file can probably be deleted, it is related to nested search which didn't ship

Comment (ijt) [reply cmd/searcher/search/eval.go]: Okay, doing that in https://github.com/sourcegraph/sourcegraph/pull/5315.

**Comment by ijt** [comment]
> These traces are useful, we are just overwhelming lightstep. Additionally we will still have the traces on the client side, so this won't stop the situation. 

We are overwhelming Lightstep by generating too many spans, IIUC. So a simple way to stop overwhelming it would be to stop doing that. Why wouldn't that work? 

> Lightstep and jaeger both have ways to handl ethis, you turn on sampling. Rather we should add a way to configure that on our larger instances.
> 
> See `./pkg/tracer`

I'm not sure what sampling means in this context. Does it mean that a random subset of requests get traces generated for them, or that within requests only a random subset of spans get generated?

I also don't know how to turn it on. I'm going to chat with the Lightstep people and see if they have some idea how to fix this.

**Comment by ijt** [comment]
This was an experimental PR anyway. Closing this in favor of something that doesn't remove information we want.

Thanks for the feedback @keegancsmith.
